### PR TITLE
fix: properly handle the tunnel error channel to retry image pushing

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This PR fixes error channel handling for Zarf tunnels so lost pod connections don't result in infinite spins.  This should mostly resolve 2104 though not marking it "Fixes" as depending on how many pod connection errors occur a deployment could still run out of retries.

## Related Issue

Relates to #2104 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/internal/packager/git/gitea.go
Comment (Noxsios) [inline src/internal/packager/git/gitea.go:1]: what happens if you are using external Gitea server?

File: src/internal/packager/images/push.go [hunk 1]
```diff
@@ -78,6 +78,14 @@ func (i *ImageConfig) PushToZarfRegistry() error {
 		defer tunnel.Close()
 	}
 
+	pushImage := func(img v1.Image, name string) error {
+		if tunnel != nil {
+			return tunnel.Wrap(func() error { return crane.Push(img, name, pushOptions...) })
+		} else {
+			return crane.Push(img, name, pushOptions...)
+		}
+	}
```
Comment (Noxsios) [inline src/internal/packager/images/push.go:87]: lint

File: src/test/e2e/main_test.go [hunk 2]
```diff
@@ -37,6 +38,9 @@ func TestMain(m *testing.M) {
 		fmt.Println(err) //nolint:forbidigo
 	}
 
+	// Set the log level to trace for when we call Zarf functions internally
+	message.SetLogLevel(message.TraceLevel)
+
```
Comment (Noxsios) [inline src/test/e2e/main_test.go:43]: ? confused what this is doing

**Review by Noxsios — CHANGES_REQUESTED** [review]
overall almost there, need to stew on if I want to advocate for `tunnel.Wrap` to take a context

File: src/internal/packager/git/gitea.go
Comment (Racer159) [reply src/internal/packager/git/gitea.go:1]: this is only for the internal Zarf-deployed one

**Review by Noxsios — APPROVED** [review]

**Pull request merged**